main.tf

```diff
@@ -40,9 +40,11 @@ resource "aws_lambda_function" "logs_to_datadog" {
 
   environment {
     variables = {
-      DD_API_KEY_SECRET_ARN = aws_secretsmanager_secret.api-key.arn
-      DD_SITE               = var.dd_site
-      DD_ENHANCED_METRICS   = var.enhanced_metrics
+      DD_API_KEY_SECRET_ARN  = aws_secretsmanager_secret.api-key.arn
+      DD_SITE                = var.dd_site
+      DD_ENHANCED_METRICS    = var.enhanced_metrics
+      DD_STORE_FAILED_EVENTS = var.store_failed_events
+      DD_S3_BUCKET_NAME      = var.bucket_name
```

Comment:
Lambda environment variables are passed in as a pair of strings so the function is likely not evaluating this to a boolean based on the code unless I'm missing something 🤔. It seems that it may be attempting to use "false" as the bucket name. 

Looking in the cloudwatch logs for the lambda (/aws/lambda/logs_to_datadog) I'm seeing this log that I suspect is being caused by this:
```
{
    "DD_EXTENSION_FAILOVER_REASON": "s3_bucket_name"
}
```
<img width="1632" alt="Screenshot 2024-08-16 at 8 21 41 AM" src="https://github.com/user-attachments/assets/47badebd-351b-483b-a31f-c296109f4a36">

We could make the extra s3 bucket, add the required lambda permissions so it can cache tags and store failed events. 

Or maybe datadog support can confirm how to disable that requirement since I'm not seeing it straightforward in the docs. Just want to make sure this doesn't have unintended consequences 🤷.

Reply:
I reached out to datadog support and they said they agree its not straightforward so theyre going to confirm internally before reaching back out to me that its a hard requirement to have the s3 bucket

Reply:
"Hi Dylan,
 
I've heard back from our Lambda specialists and they've confirmed that the S3 bucket is also required for caching tags for logs, it is not only used for storing failed events so it is still necessary even if DD_STORE_FAILED_EVENTS is set to false. In the error snippet you can see multiple mentions of attempting to store tags in the S3 cache.
 
My apologies for not having confirmed this sooner, but it seems that this env variable is necessary for the Lambda Log Forwarder. "

im looking through dev, I dont really see a bucket where I think this would make sense unless you have a suggestion  @magreenbaum ill just add a new bucket to this module and store everything there 

theres a "datadog-serverless-forwarderzipsbucket-1ae71q0dw8myq" bucket but I think that was apart of the old configuration when cloudformation was used and its not replicated in qa or prod

Reply:
I think a new bucket makes the most sense. 👍

Reply:
added the S3 bucket from the S3 repo as well as used the kms key created in the module to encrypt the objects, added some permissions to that role to allow the datadog lambda to pull tags from resources